lib/srv/sess.go

```diff
@@ -1026,6 +1069,15 @@ func (s *session) startInteractive(ch ssh.Channel, ctx *ServerContext) error {
 
 	ctx.Debug("Waiting for continue signal")
 
+	sessionUser, err := user.Lookup(ctx.Identity.Login)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if tempUser != nil && sessionUser.Uid != tempUser.Uid {
+		return trace.Errorf("UID changed between session start and terminal start: start(%s) != now(%s)",
+			tempUser.Uid, sessionUser.Uid)
```

Comment:
Here and in `startExec` we could do the `user.Lookup` part only when `tempUser != nil`, so we avoid adding a potential failure case when we don't actually need to check anything.